test/integration/camera.cc

```diff
@@ -293,5 +296,130 @@ TEST_P(CameraSensorTest, LInt8ImagesWithBuiltinSDF)
+  if (!engine)
+  {
+    gzdbg << "Engine '" << _renderEngine
+          << "' is not supported" << std::endl;
+    return;
+  }
+
+  gz::rendering::ScenePtr scene = engine->CreateScene("scene");
+
+  // Do the test
+  gz::sensors::Manager mgr;
+
+  auto *sensor1 =
+      mgr.CreateSensor<gz::sensors::CameraSensor>(cameraWithoutIntrinsicsTag);
+  auto *sensor2 =
+      mgr.CreateSensor<gz::sensors::CameraSensor>(cameraWithIntrinsicsTag);
+  ASSERT_NE(sensor1, nullptr);
+  ASSERT_NE(sensor2, nullptr);
+  sensor1->SetScene(scene);
+  sensor2->SetScene(scene);
+
+  std::string imageTopic1 = "/camera1/image";
+  std::string infoTopic1 = "/camera1/camera_info";
+  std::string imageTopic2 = "/camera2/image";
+  std::string infoTopic2 = "/camera2/camera_info";
+  WaitForMessageTestHelper<gz::msgs::Image> helper1(imageTopic1);
+  WaitForMessageTestHelper<gz::msgs::CameraInfo> helper2(infoTopic1);
+  WaitForMessageTestHelper<gz::msgs::Image> helper3(imageTopic2);
+  WaitForMessageTestHelper<gz::msgs::CameraInfo> helper4(infoTopic2);
+  EXPECT_TRUE(sensor1->HasConnections());
+  EXPECT_TRUE(sensor2->HasConnections());
+
+  // Update once to create image
+  mgr.RunOnce(std::chrono::steady_clock::duration::zero());
+
+  EXPECT_TRUE(helper1.WaitForMessage()) << helper1;
+  EXPECT_TRUE(helper2.WaitForMessage()) << helper2;
+  EXPECT_TRUE(helper3.WaitForMessage()) << helper3;
+  EXPECT_TRUE(helper4.WaitForMessage()) << helper4;
+
+  // Subscribe to the camera info topic
+  gz::msgs::CameraInfo camera1Info, camera2Info;
+
+  std::function<void(const gz::msgs::CameraInfo&)> camera1InfoCallback =
+      [&camera1Info](const gz::msgs::CameraInfo& _msg) {
+        camera1Info = _msg;
+  };
+
+  std::function<void(const gz::msgs::CameraInfo&)> camera2InfoCallback =
+      [&camera2Info](const gz::msgs::CameraInfo& _msg) {
+        camera2Info = _msg;
+  };
+
+  // Subscribe to the camera topic
+  gz::transport::Node node;
+  node.Subscribe(infoTopic1, camera1InfoCallback);
+  node.Subscribe(infoTopic2, camera2InfoCallback);
+
+  // Wait for a few camera frames
+  mgr.RunOnce(std::chrono::steady_clock::duration::zero(), true);
+
+  // Run to get image and check image format in callback
+  int sleep = 0;
+  int maxSleep = 10;
+  while (sleep++ < maxSleep)
+  {
+    std::lock_guard<std::mutex> lock(g_mutex);
+    std::this_thread::sleep_for(std::chrono::milliseconds(100));
+  }
+
+  // Image size, focal length and optical center
+  // Camera sensor without intrinsics tag
+  double error = 1e-1;
+  EXPECT_DOUBLE_EQ(camera1Info.width(), 1000);
+  EXPECT_DOUBLE_EQ(camera1Info.height(), 1000);
+  EXPECT_NEAR(camera1Info.intrinsics().k(0), 863.2297, error);
+  EXPECT_NEAR(camera1Info.intrinsics().k(4), 863.2297, error);
+  EXPECT_DOUBLE_EQ(camera1Info.intrinsics().k(2), 500);
+  EXPECT_DOUBLE_EQ(camera1Info.intrinsics().k(5), 500);
+
+  // Camera sensor with intrinsics tag
+  EXPECT_DOUBLE_EQ(camera2Info.width(), 1000);
+  EXPECT_DOUBLE_EQ(camera2Info.height(), 1000);
```

Comment:
I think we are comparing unsigned integers, instead of doubles
```suggestion
  EXPECT_EQ(camera2Info.width(), 1000u);
  EXPECT_EQ(camera2Info.height(), 1000u);
```